extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java

```diff
@@ -442,7 +441,7 @@ public void uploadBlockBlob(
       blobContainerClient
           // Creates a blob by default, no need to use a specific blob client.
           // We also need to urlEncode the path to handle special characters.
-          .getBlobClient(Utility.urlEncode(blobName))
+          .getBlobClient(blobName)
```

Comment:
Since you have a test setup running, could you please try with other special characters like 
@, / ,     , _, - * .